### PR TITLE
DynamoDB configurations update

### PR DESCRIPTION
**Pull request by marcinczeczko**
- Refactored configuration structure - broken down config into dynamodb, AWS, SDK & HTTP client groups
- Added config of TLS Managers for sync/async clients
- Added config of HTTP Proxy for async client
- Added support for SDK client interceptors

### CODE REVIEW

**Comment by marcinczeczko** [comment]
@gsmet - I updated the extension due to the new config options available in latest AWS SDK: proxy for async, tlsmanagers, interceptors.

File: extensions/amazon-dynamodb/runtime/src/main/java/io/quarkus/dynamodb/runtime/NettyHttpClientConfig.java
Comment (machi1990) [inline extensions/amazon-dynamodb/runtime/src/main/java/io/quarkus/dynamodb/runtime/NettyHttpClientConfig.java]: `defaultValue` is not needed, boolean default to `false`

File: extensions/amazon-dynamodb/runtime/src/main/java/io/quarkus/dynamodb/runtime/SdkConfig.java
Comment (marcinczeczko) [reply extensions/amazon-dynamodb/runtime/src/main/java/io/quarkus/dynamodb/runtime/SdkConfig.java]: Fixed. Additionally migrated to the config docs automatically generated.

File: extensions/amazon-dynamodb/runtime/src/main/java/io/quarkus/dynamodb/runtime/NettyHttpClientConfig.java
Comment (marcinczeczko) [reply extensions/amazon-dynamodb/runtime/src/main/java/io/quarkus/dynamodb/runtime/NettyHttpClientConfig.java]: I left it to let the doc generator to show default value

File: docs/src/main/asciidoc/dynamodb-guide.adoc
Comment (machi1990) [inline docs/src/main/asciidoc/dynamodb-guide.adoc]: +1 :-)

**Comment by marcinczeczko** [comment]
btw, I opened quickstarts PR https://github.com/quarkusio/quarkus-quickstarts/pull/282 to align with this update.

File: extensions/amazon-dynamodb/runtime/src/main/java/io/quarkus/dynamodb/runtime/ApacheHttpClientConfig.java
Comment (gsmet) [inline extensions/amazon-dynamodb/runtime/src/main/java/io/quarkus/dynamodb/runtime/ApacheHttpClientConfig.java]:
```suggestion
     * Each built HTTP client has its own private connection pool.
```

File: extensions/amazon-dynamodb/runtime/src/main/java/io/quarkus/dynamodb/runtime/ApacheHttpClientConfig.java
Comment (gsmet) [inline extensions/amazon-dynamodb/runtime/src/main/java/io/quarkus/dynamodb/runtime/ApacheHttpClientConfig.java]: I wonder if we should make them hard defaults so that they are properly documented. WDYT?

File: extensions/amazon-dynamodb/runtime/src/main/java/io/quarkus/dynamodb/runtime/ApacheHttpClientConfig.java
Comment (gsmet) [inline extensions/amazon-dynamodb/runtime/src/main/java/io/quarkus/dynamodb/runtime/ApacheHttpClientConfig.java]: In straight Javadoc, you would need a `<p>`, wouldn't you? Or the doc output is OK?

**Review by gsmet — CHANGES_REQUESTED** [review]
I spotted a typo and added a comment and a question. Can you have a look?

We're nearly there :).

File: extensions/amazon-dynamodb/runtime/src/main/java/io/quarkus/dynamodb/runtime/ApacheHttpClientConfig.java
Comment (marcinczeczko) [reply extensions/amazon-dynamodb/runtime/src/main/java/io/quarkus/dynamodb/runtime/ApacheHttpClientConfig.java]: Make sense, but I'm wondering now if I should document it in here. It's basically because those default are buried deep in the SDK builders and took me some time to find them out. So I think I should rather remove them. WDYT ?

File: extensions/amazon-dynamodb/runtime/src/main/java/io/quarkus/dynamodb/runtime/ApacheHttpClientConfig.java
Comment (marcinczeczko) [reply extensions/amazon-dynamodb/runtime/src/main/java/io/quarkus/dynamodb/runtime/ApacheHttpClientConfig.java]: Yes I missed that. Fixed.

Comment (marcinczeczko) [reply extensions/amazon-dynamodb/runtime/src/main/java/io/quarkus/dynamodb/runtime/ApacheHttpClientConfig.java]: I mean added paragraph to the next row. Basically the doc output when checking the generated HTML is fine

**Comment by gsmet** [comment]
@marcinczeczko I rebased and I added a couple of commits to improve auto-documentation. Can you have a look at them?

I'll wait for your blessing before merging.

**Review by gsmet — APPROVED** [review]

**Comment by marcinczeczko** [comment]
Thanks for those updates, looks great. You have my blessing :)

**Comment by gsmet** [comment]
Perfect, thanks!

Kudos for this work. I will merge this and check the quickstart update tomorrow.

**Pull request merged**
